src/lib/media-query/observable-media.spec.ts

```diff
@@ -71,13 +68,16 @@ describe('observable-media', () => {
         [ObservableMedia, MatchMedia],
         (mediaService, matchMedia) => {
           let count = 0,
-              subscription = mediaService
-                  .asObservable()
-                  .filter(change => change.mqAlias == 'md')
-                  .map(change => change.mqAlias)
-                  .subscribe((alias) => {
-                    count += 1;
-                  });
+              subscription = map.call(
```

Comment:
Yes. I had an RxChain usage... but it was clumsy and not elegant. I will need your help to review that solution.

```
// Use non-prototype-patching Rx Operators
import {fromEvent} from 'rxjs/observable/fromEvent';
import {RxChain, takeUntil, map, switchMap} from '@angular/cdk';

export class SplitHandleDirective {

  @Output() drag: Observable<{ x: number, y: number }>;

  constructor(ref: ElementRef) {
    const mousedown$ = fromEvent.call(ref.nativeElement, 'mousedown');
    const mousemove$ = map.call(fromEvent.call(document, 'mousemove'), getMouseEventPosition);  // tslint:disable-line:max-line-length
    const mouseup$ = fromEvent.call(document, 'mouseup');

    this.drag = RxChain.from(mousedown$)
        .call(map, getMouseEventPosition)
        .call(switchMap, () => {
            return takeUntil.call(mousemove$, mouseup$);
        }).result() as Observable<{ x: number, y: number }>;
  }
}
```